Assemble weights separately from the design matrix

wcwitt opened the pull request:
Hi @cortner, one option towards https://github.com/ACEsuit/ACE1pack.jl/issues/125 is to assemble the full weight vector in a separate step from the design matrix. I've started to sketch it out - before I do anything else, does this direction sound good for the short term? It wouldn't preclude anything else we've discussed over longer timescales.

cortner commented:
I've merged main into this branch and made some small fixes - this now works with tutorials. If you are happy then can we merge and tag?

wcwitt commented:
Thanks - I fixed that same bug once, but it was on a plane and apparently I never pushed. At the same time, I updated the ACE1pack tests/tutorials to work with assemble returning A,Y instead of A,Y,W. But I can understand your point about that being a big change. Let's leave things as the are now. Merging.
